pkg/config/setup/system_probe.go

```diff
@@ -241,7 +241,7 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 	cfg.BindEnv(join(smNS, "tls", "native", "enabled"))
 
 	// For backward compatibility
-	cfg.BindEnv(join(smNS, "enable_go_tls_support"))
+	cfg.BindEnvAndSetDefault(join(smNS, "enable_go_tls_support"), true)
 	cfg.BindEnv(join(smNS, "tls", "go", "enabled"))
```

Comment:
Since we have backward compatibility, the change should be done in `adjust_usm`
Only if the flags weren't set, then we should set to true